### PR TITLE
chore: update libp2p-pubsub

### PR DESCRIPTION
**Pull request by wemeetagain**
Update `libp2p-pubsub` to `v0.6.x` (passes tests with libp2p/js-libp2p-pubsub#65 and  libp2p/js-libp2p-pubsub#66 applied)

- Update `InMessage` object to include `receivedFrom` peer-id string
  - Update methods that previously took `(id: string, msg: InMessage)` to just be `(msg: InMessage)` since InMessage now contains the `receivedFrom` peer id
- Update `validate` to throw on invalid messages
  - Remove `_processTopicValidatorResult` in favor of overriding `validate` in order to apply validation penalties
- Update `publish`/`_publish` for publishing a _single_ message
- Update tests as needed, remove now irrelevant tests
- Replace `peer`/`Peer` with `peerStreams`/`PeerStreams` in many places
  - (in a future PR, will refactor gossipsub maps/internal methods to use peer id string identifiers instead of `peerStreams` in many places)
- Remove special handling from https://github.com/ChainSafe/js-libp2p-gossipsub/pull/39 which isn't needed. If a subscription is triggered after a peer has opened an inbound stream but before a peer has opened an outbound stream, we don't need special handling. The subscription will simply be sent when the peer opens the outbound stream (via `_onPeerConnected`).

### CODE REVIEW

File: ts/index.ts [hunk 1]
```diff
@@ -905,148 +904,155 @@ class Gossipsub extends BasicPubsub {
...
+          }
+        })
+      } else {
+        // non-flood-publish behavior
+        // send to direct peers, subscribed floodsub peers
+        // and some mesh peers above publishThreshold
+
+        // direct peers
+        this.direct.forEach(id => {
+          const peer = this.peers.get(id)
```
Comment (vasco-santos) [inline ts/index.ts:942]: peerStreams

File: ts/pubsub.js [hunk 2]
```diff
@@ -159,8 +167,8 @@ class BasicPubSub extends Pubsub {
 
     if (msgs.length) {
       msgs.forEach(async message => {
```
Comment (vasco-santos) [inline ts/pubsub.js:169]: do we need this async?

File: ts/pubsub.js [hunk 3]
```diff
@@ -132,24 +129,35 @@ class BasicPubSub extends Pubsub {
...
     const subs = rpc.subscriptions
     const msgs = rpc.msgs
 
     if (subs.length) {
       // update peer subscriptions
-      peer.updateSubscriptions(subs)
-      subs.forEach((subOpt) => this._processRpcSubOpt(peer, subOpt))
-      this.emit('pubsub:subscription-change', peer.id, peer.topics, subs)
+      subs.forEach((subOpt) => this._processRpcSubOpt(idB58Str, subOpt))
+      this.emit('pubsub:subscription-change', peerStreams.id, subs)
```
[the remark this answers is not in the document]
Comment (wemeetagain) [reply ts/pubsub.js:160]: We aren't indexing "peer => subscribed topics" anymore.
Previously the `Peer` maintained a list of topics - `Peer#topics`
Now we just have `this.topics` - topic => subscribed peers

we could:
1. leave this as is, keeping this event modified
2. loop thru all this.topics, collecting subscribed peers
3. create another index, eg: `this.peerTopics` - peer => subscribed topics

I've got a preference for 1 or 3

[the remark this answers is not in the document]
Comment (vasco-santos) [reply ts/pubsub.js:160]: Let's leave it for now. If we have something built on top of this information, we can reconsider what to do

**Comment by vasco-santos** [comment]
CI seems broken. Any cache related issue?

**Comment by wemeetagain** [comment]
Yeah, just cleared the cache and rerunning now

**Comment by wemeetagain** [comment]
It seems the tests are failing w floodsub interop. Floodsub seems to be using the 0.6.x branch of libp2p-pubsub :thinking:

**Comment by wemeetagain** [comment]
I'll put in a PR to update floodsub, link to that branch here

**Pull request merged**